src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -68,5 +96,287 @@ protected static SymbolDisplayPart NewLine()
+            {
+                var parameterIndex = argToParamMap[argumentIndex];
+                if (parameterIndex < 0)
+                {
+                    continue;
+                }
+
+                var parameter = parameters[parameterIndex];
+                var argument = arguments[argumentIndex];
+
+                if (!IsCompatibleArgument(argument, parameter))
+                {
+                    foundParameterIndex = -1;
+                    argToParamMap.Free();
+                    return false;
+                }
+            }
+
+            // find the parameter at the cursor position
+            var argumentIndexToSave = TryGetArgumentIndex(arguments, position);
+            if (argumentIndexToSave >= 0)
+            {
+                var foundParam = argToParamMap[argumentIndexToSave];
+                if (foundParam >= 0)
+                {
+                    foundParameterIndex = foundParam;
+                }
+                else
+                {
+                    var firstUnspecified = FirstUnspecifiedParameter(argToParamMap, argumentCount);
+                    foundParameterIndex = firstUnspecified < 0 ? 0 : firstUnspecified;
+                }
+            }
+            else
+            {
+                foundParameterIndex = argumentIndexToSave >= 0 ? argToParamMap[argumentIndexToSave] : -1;
+            }
+
+            Debug.Assert(foundParameterIndex < parameters.Length);
+
+            argToParamMap.Free();
+            return true;
+
+            // If the cursor is pointing at an argument for which we did not find the corresponding
+            // parameter, we will highlight the first unspecified parameter.
+            static int FirstUnspecifiedParameter(ArrayBuilder<int> argToParamMap, int argumentCount)
+            {
+                var specified = ArrayBuilder<bool>.GetInstance(argumentCount, false);
+                for (var i = 0; i < argumentCount; i++)
+                {
+                    var parameterIndex = argToParamMap[i];
+                    if (parameterIndex >= 0)
+                    {
+                        specified[parameterIndex] = true;
+                    }
+                }
+
+                var first = specified.FindIndex(s => !s);
+                specified.Free();
+                return first;
+            }
+
+            // Determines if the given argument is compatible with the given parameter
+            bool IsCompatibleArgument(ArgumentSyntax argument, IParameterSymbol parameter)
+            {
+                var parameterRefKind = parameter.RefKind;
+                if (parameterRefKind == RefKind.None)
+                {
+                    if (IsEmptyArgument(argument.Expression))
+                    {
+                        return true;
+                    }
+
+                    var type = parameter.Type;
+                    if (parameter.IsParams && HasImplicitConversion(argument.Expression, ((IArrayTypeSymbol)type).ElementType))
```

Comment:
check that the tyep is actually an array first (i don't see why it would have to be in the case of errors.  for example `params int x`.

Reply:
Good catch. That was crashing